libraries/AP_InertialSensor/AP_InertialSensor_ISM330DHCX.cpp

```diff
@@ -0,0 +1,362 @@
+    }
+
+
+    uint8_t fifo_ctrl3 = 0x66; //417Hz for both accel and gyro
+    _dev->write_register(ISM330DHCX_ADDR_FIFO_CTRL3, fifo_ctrl3);
+
+    hal.scheduler->delay(5);
+
+    _dev->write_register(ISM330DHCX_ADDR_FIFO_CTRL4, 0x06); // Continuous mode: if the FIFO is full, the new sample overwrites the older one;
+    hal.scheduler->delay(5);
+
+
+    _dev->write_register(ISM330DHCX_ADDR_CTRL1_XL, 0x6C);  //416 Hz  without filtering
+
+    hal.scheduler->delay(5);
+
+    _dev->write_register(ISM330DHCX_ADDR_CTRL2_G, 0x6C); // 
+    hal.scheduler->delay(5);
+
+    _dev->write_register(ISM330DHCX_ADDR_CTRL3_C, 0x04); //
+
+    //Filter configuration
+
+    uint8_t ctrl_reg7g = 0x44; //0100 0100
+    _dev->write_register(ISM330DHCX_ADDR_CTRL7_G, ctrl_reg7g);
+
+    hal.scheduler->delay(5);
+
+
+
+    // Set up the filter desired
+    uint16_t filter_hz = _accel_filter_cutoff();
+    _set_filter_frequency(filter_hz);
+
+    _dev->get_semaphore()->give();
+    
+    return true;
+}
+
+bool AP_InertialSensor_ISM330DHCX::_gyro_init()
+{
+
+    return true;
+}
+
+bool AP_InertialSensor_ISM330DHCX::_init_sensor(void)
+{
+    
+    _accel_init();
+
+    _gyro_init();
+
+    return true;
+}
+
+/*
+  startup the sensor
+ */
+void AP_InertialSensor_ISM330DHCX::start(void)
+{
+    if (!_imu.register_gyro(gyro_instance, 800, _dev->get_bus_id_devtype(DEVTYPE_INS_ISM330DHCX))) {
+        return;
+    }
+
+    if (!_imu.register_accel(accel_instance, 800, _dev->get_bus_id_devtype(DEVTYPE_INS_ISM330DHCX))) {
+        return;
+    }
+
+    set_accel_orientation(accel_instance, _rotation);
+    set_gyro_orientation(gyro_instance, _rotation);
+    // start the timer process to read samples
+    _dev->register_periodic_callback(1250, FUNCTOR_BIND_MEMBER(&AP_InertialSensor_ISM330DHCX::_accumulate_accel, void));
+}
+
+
+/*
+  set the filter frequency
+ */
+void AP_InertialSensor_ISM330DHCX::_set_filter_frequency(uint8_t filter_hz)
+{
+    _accel_filter.set_cutoff_frequency(400, filter_hz);
+    _gyro_filter.set_cutoff_frequency(400, filter_hz);
+}
+
+
+/*
+  copy filtered data to the frontend
+ */
+bool AP_InertialSensor_ISM330DHCX::update(void)
+{
+    update_gyro(gyro_instance);
+    update_accel(accel_instance);
+
+    return true;
+}
+
+// Accumulate values from accels and gyros
+void AP_InertialSensor_ISM330DHCX::_accumulate_gyro (void)
+{
+
+}
+
+void AP_InertialSensor_ISM330DHCX::_accumulate_accel (void)
+{
+    uint8_t num_samples_available;
+    uint8_t fifo_status = 0;
+    //uint8_t fifo_tag = 0;
+    uint8_t status = 0;
+
+    // Read accelerometer FIFO to find out how many samples are available
+    _dev->read_registers(ISM330DHCX_ADDR_FIFO_STATUS1,
+                         &fifo_status, 1);
+    num_samples_available = fifo_status;
+
+    _dev->read_registers(ISM330DHCX_ADDR_STATUS_REG,
+                         &status, 1);
+
+    // read the samples and apply the filter
+    if ((num_samples_available > 0) || (status)) {
+        if (status & 0x01) {
```

Comment:
Soem defines for these bits would be good. eg, `ISM330DHCX_ACC_SAMPLES_AVAILABLE`?